core/src/main/java/org/apache/iceberg/view/ViewOperations.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import org.apache.iceberg.io.FileIO;
+
+/** SPI interface to abstract view metadata access and updates. */
+public interface ViewOperations {
+
+  /**
+   * Return the currently loaded view metadata, without checking for updates.
+   *
+   * @return view metadata
+   */
+  ViewMetadata current();
+
+  /**
+   * Return the current view metadata after checking for updates.
+   *
+   * @return view metadata
+   */
+  ViewMetadata refresh();
+
+  /**
+   * Replace the base view metadata with a new version.
+   *
+   * <p>This method should implement and document atomicity guarantees.
+   *
+   * <p>Implementations must check that the base metadata is current to avoid overwriting updates.
+   * Once the atomic commit operation succeeds, implementations must not perform any operations that
+   * may fail because failure in this method cannot be distinguished from commit failure.
+   *
+   * <p>Implementations must throw a {@link
+   * org.apache.iceberg.exceptions.CommitStateUnknownException} in cases where it cannot be
+   * determined if the commit succeeded or failed. For example if a network partition causes the
+   * confirmation of the commit to be lost, the implementation should throw a
+   * CommitStateUnknownException. This is important because downstream users of this API need to
+   * know whether they can clean up the commit or not, if the state is unknown then it is not safe
+   * to remove any files. All other exceptions will be treated as if the commit has failed.
+   *
+   * @param base view metadata on which changes were based
+   * @param metadata new view metadata with updates
+   */
+  void commit(ViewMetadata base, ViewMetadata metadata);
+
+  /** Returns a {@link FileIO} to read and write view files. */
+  FileIO io();
```

Comment:
Why does this need to supply a `FileIO`? Engines should not be reading files from what is in the view, so I don't think it is necessary to expose one. It's not in the `View` API.

Reply:
we're using `io()` mainly to read+write the view metadata in `BaseViewOperations`. Are you suggesting we should remove `io()` from `ViewOperations` and rather add `protected abstract FileIO io()` to `BaseViewOperations`?

Reply:
Yes. This should not be public.